### PR TITLE
Changed default nccl timeout to 600s issue #2703

### LINKED ISSUE #2703: NCCL timeout remains at 600s and does not get updated if 1800s passed when initializing Accelerator
### System Info

```Shell
Accelerate version: 0.30.0.dev0
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

When doing 
```
init_proc = InitProcessGroupKwargs(timeout=timedelta(seconds=1800))
accelerator = Accelerator(log_with="wandb", kwargs_handlers=[init_proc])
```
This won't update the timeout [here](https://github.com/huggingface/accelerate/blob/f7daaaa305362e1a959183e23bf85244798ceb35/src/accelerate/state.py#L213).
Why? Because the `kwargs` are generated by `init_proc.get_kwargs()` which returns `{}` if the timeout is the defualt. However, for `torch.distributed.init_process_group` the default timeout **is not** 1800s.

So maybe change the default in `InitProcessGroupKwargs` to 600?

### Expected behavior

It should set the timeout correctly.

### PR DESCRIPTION
**Pull request by jubueche**
# What does this PR do?

Fixes #2703 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
Discussed [here](https://github.com/huggingface/accelerate/issues/2703).
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/accelerate/utils/dataclasses.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ class InitProcessGroupKwargs(KwargsHandler):
 
     backend: Optional[str] = "nccl"
     init_method: Optional[str] = None
-    timeout: timedelta = timedelta(seconds=1800)
+    timeout: timedelta = timedelta(seconds=600)
```
Comment (muellerzr) [inline src/accelerate/utils/dataclasses.py:169]:
```suggestion
    timeout: Optional[timedelta] = None
```

This actually gets a bit more nuanced. Let's change this to `None`, and then do the following in a `__post_init__`:

```python
if self.timeout is None:
    seconds = 1800 if self.backend != "nccl" else 600
    self.timeout = timedelta(seconds=seconds)
```

This comes directly from the torch docs:
> Default value is 10 minutes for NCCL and 30 minutes for other backends.

**Review by muellerzr — COMMENTED** [review]
Thanks! Let's be a bit more thorough though please :)

**Review by SunMarc — COMMENTED** [review]
+1 with @muellerzr comments. We could potentially add a description for `timeout` since I feel time `timeout` is used a lot and now it is kind of complicated since the default value depends on the backend.

**Comment by SunMarc** [comment]
Closing this PR since the PR above fixes the issue !

**Pull request closed without merging**